src/mkdocs_autorefs/plugin.py

```diff
@@ -133,7 +146,9 @@ def on_config(self, config: MkDocsConfig) -> MkDocsConfig | None:
             The modified config.
         """
         log.debug("Adding AutorefsExtension to the list")
-        config["markdown_extensions"].append(AutorefsExtension())
+        scan_anchors = self.scan_anchors or self.config.scan_anchors
+        anchor_scanner_factory = partial(AnchorScannerTreeProcessor, self) if scan_anchors else None
+        config["markdown_extensions"].append(AutorefsExtension(anchor_scanner_factory))
```

Comment:
Maybe we could simply accept a plugin instance directly in `AutorefsExtension`:

```suggestion
        config["markdown_extensions"].append(AutorefsExtension(plugin=self if scan_anchors else None))
```

Reply:
yes